optimade/client/utils.py

```diff
@@ -116,6 +116,10 @@ def __init__(self):
             refresh_per_second=10,
         )
 
+    def print(self, *args, **kwargs):
+        if not self.disable:
+            super().print(*args, **kwargs)
+
```

Comment:
At a yield statement, the execution of the function is paused until the function is called again. 
So in the silent_raise() function below(not part of this PR), sys.tracebacklimit and sys.excepthook are only reset the second time the silent_raise() function is called. I do not think this is the behaviour you want.